### PR TITLE
Update Welcome Guides to use external image URLs

### PR DESCRIPTION
**Pull request by noisysocks**
## Description

Updates the welcome guides in `edit-post`, `edit-widgets` and `customize-widgets` so that they use images that are served from `wp-admin/images/block-editor/*` instead of `data:` URLs that are included in the JavaScript bundle. This should significantly reduce the bundle sizes of `edit-post`, `edit-widgets` and `customize-widgets.`

https://github.com/WordPress/wordpress-develop/pull/1269 is the Core PR which adds the images to `wp-admin/images/block-editor`.

## How has this been tested?

You'll be testing https://github.com/WordPress/wordpress-develop/pull/1269 and this PR at the same time.

1. Clone the `wordpress-develop` repo, check out https://github.com/WordPress/wordpress-develop/pull/1269, build using `npm i; grunt build`.
2. Clone the `gutenberg` repo, check out https://github.com/WordPress/gutenberg/pull/32026, build using `npm i; npm run build`.
3. Configure your environment to run both local copies. For example, if using `wp-env`, modify `gutenberg/.wp-env.json` to set `"core": "/path/to/wordpress-develop/build"`, then run `wp-env start`.
4. Go to Posts → Add New, open the Options menu, and click Welcome Guide. The guide's images should appear normally.
5. Go to Appearance → Widgets, open the Options menu, and click Welcome Guide. The guide's images should appear normally.
6. Go to Appearance → Customize → Widgets, open the Options menu, and click Welcome Guide. The guide's image should appear normally.

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by noisysocks** [comment]
https://github.com/WordPress/gutenberg/pull/32026#issuecomment-844709178 is what I wanted to see! 🙂

@youknowriad: Any concerns with this approach?

I'm happy with it but I think the `src` needs to fallback to `wp-content/plugins/gutenberg/images/*` or maybe a `w.org` URL if the user is running a version of WordPress that doesn't contain https://github.com/WordPress/wordpress-develop/pull/1269. Maybe this can be done with `<picture>`.

**Comment by kevin940726** [comment]
I don't understand why we have to add the images to core? Couldn't we serve the images from this plugin instead? Rather than using data url, is it possible to serve the images via endpoints specifically for them?

**Comment by talldan** [comment]
There's a welcome guide in the template editor now as well that's worth testing - https://github.com/WordPress/gutenberg/pull/31330

**Comment by youknowriad** [comment]
Oh wow, this is definitely promising. 

What I don't like is the magic URLs, it's a package so there's no guarantee the image exist like that with that relative URL. I wonder if we should do the same thing like previews/patterns. This is upload the images to wp.org (I believe this requires a meta ticket) and just use the external URLs here.

**Comment by noisysocks** [comment]
> I don't understand why we have to add the images to core? Couldn't we serve the images from this plugin instead?

Yes but eventually `@wordpress/edit-post` etc. will be published to npm and used by WordPress installations that don't have the Gutenberg plugin installed.

> What I don't like is the magic URLs, it's a package so there's no guarantee the image exist like that with that relative URL. I wonder if we should do the same thing like previews/patterns. This is upload the images to wp.org (I believe this requires a meta ticket) and just use the external URLs here.

We have this problem with relative URLs in a few other places, too. I was hesitant to use `w.org` because it means that this feature won't work if you're using WordPress locally without an internet connection. But I suppose that's not a very practical concern and `w.org` solves a lot of our problems, so I'll go down this path 😀

**Comment by noisysocks** [comment]
Meta ticket: https://meta.trac.wordpress.org/ticket/5741

**Comment by gziolo** [comment]
Cool, https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-post/src/components/welcome-guide/images.js is 712 kB on the disk and it has to be downloaded, parsed and executed even when someone disabled the welcome guide 😱 

It looks like there is also a notice from Babel:

> [BABEL] Note: The code generator has deoptimised the styling of /tmp/7eb1a89b-1d08-4e73-a1a6-a70f98ec657e/packages/edit-post/src/components/welcome-guide/images.js as it exceeds the max of 500KB.

**Comment by mcsf** [comment]
> Meta ticket: https://meta.trac.wordpress.org/ticket/5741

Now that that ticket is closed, is the only thing left to do to update the `src` paths in this PR to `s.w.org`?

**Comment by Mamaduka** [comment]
We also need to add Template Editor/Template Mode graphics to the `s.w.org`. See #32055.

**Comment by Mamaduka** [comment]
It looks like all assets are on `s.w.org`.

**Comment by noisysocks** [comment]
Thanks @Mamaduka. I've rebased this and updated `TemplateEditorImage` to use the external URL.

**Comment by noisysocks** [comment]
I've updated everything to use the `s.w.org` URLs. This should be ready to go now.

File: packages/customize-widgets/src/components/welcome-guide/index.js [hunk 1]
```diff
@@ -22,7 +21,20 @@ export default function WelcomeGuide( { sidebar } ) {
 
 	return (
 		<div className="customize-widgets-welcome-guide">
-			<EditorImage />
+			<img
```
Comment (kevin940726) [inline packages/customize-widgets/src/components/welcome-guide/index.js]: Why inlining all these images when we can just change the `src` attributes?

**Comment by kevin940726** [comment]
Not related to this PR, but now that we drop support for IE, I think we can stop using the `display: none` hack and just use `<picture>` element? Here's a great [article](https://css-tricks.com/gifs-and-prefers-reduced-motion/) about it.

Furthermore, the [`welcome-canvas.gif`](https://s.w.org/images/block-editor/welcome-canvas.gif) file weighs about 82 kB. Would it be possible to convert it to a video (mp4 or other formats) file to further reduce the size? Even though it only shows once, it's still one of the first resource the editor loads. It'd be nice if we can optimize it somehow (in another PR of course).

File: packages/customize-widgets/src/components/welcome-guide/index.js (hunk 1 above)
Comment (noisysocks) [reply packages/customize-widgets/src/components/welcome-guide/index.js]: I figured there's no real benefit to having a seperate `images.js` file now that there's no massive lines of base64 data.

**Comment by noisysocks** [comment]
> Not related to this PR, but now that we drop support for IE, I think we can stop using the display: none hack and just use <picture> element? Here's a great article about it.

This is a really good idea! Done in 18c75fa.

> Furthermore, the welcome-canvas.gif file weighs about 82 kB. Would it be possible to convert it to a video (mp4 or other formats) file to further reduce the size? Even though it only shows once, it's still one of the first resource the editor loads. It'd be nice if we can optimize it somehow (in another PR of course).

Agree this would be nice. Let's do it later though.

File: packages/customize-widgets/src/components/welcome-guide/style.scss [hunk 2]
```diff
@@ -7,20 +7,6 @@
 
 	&__image {
 		height: 240px;
```
Comment (kevin940726) [inline packages/customize-widgets/src/components/welcome-guide/style.scss]: Do we still need this?

**Review by kevin940726 — APPROVED** [review]
This is looking better and cleaner! Let's iterate other optimizations later.

File: packages/customize-widgets/src/components/welcome-guide/style.scss (hunk 2 above)
Comment (noisysocks) [reply packages/customize-widgets/src/components/welcome-guide/style.scss]: Nope!

**Pull request merged**